src/main/groovy/com/athaydes/spockframework/report/extension/SpockReportsSpecificationExtension.groovy

```diff
@@ -51,7 +51,7 @@ class InfoContainer {
     private static String keyFor( String specName,
                                   FeatureInfo feature,
                                   IterationInfo iteration ) {
-        def index = Utils.isUnrolled( feature ) ? iteration.iterationIndex : -1
+        def index = Utils.isUnrolled( feature ) || ( iteration != null && iteration.estimatedNumIterations < 0 ) ? iteration.iterationIndex : -1
```

Comment:
This is the "rule" I come out with after debugging datasets with @Unrolled annotation or not: if there is @Unrolled, then this will be positive number same as datasets number, while if not, this will be -1. I agree this is not "safe", and I couldn't find doc explaining this behavior or saying this is expected. Also it is counter intuitive when seeing number of iteration being negative. But this is the only difference I spot for these 2 cases at the moment

Reply:
The `Utils.isUnrolled` method is supposed to tell whether the feature was unrolled. I think the logic became insufficient after a few new Spock releases (unrolled has become the default, I think?!)... I will look into this a bit more.

Reply:
from 2.0, indeed, it should be unroll by default, https://github.com/spockframework/spock/issues/967, probably there is changes in this from 3.0, thus leading to this issue